### PR TITLE
Cds 67121 cds 71172 post prod rollback

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by piyush-bhuwalka — APPROVED** [review]

File: docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* Rollback deployment is currently supported for the following deployment types only: 
+  - Kubernetes
+  - Tanzu Application Services (TAS)
+
+    Harness anticipates expanding this feature to other deployment types in the future.
+* You can rollback successful pipelines only. The rollback option is not available for failed pipelines.
+* Only the rollback steps that are part of the stage can be rolled back.
+* You cannot rollback the same pipeline multiple times. It can be rolled back if the pipeline is executed again.
+* If the pipeline configuration has changed between executions, the previous execution YAML is used as reference to rollback.
+* If the rollback steps of a stage have the same expressions present in the steps of a successful deployment, the steps of the deployed pipeline must be resolved first. For example, you should be able to reuse a step in the rollback section as the output of a step in the steps section.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md:26]: @sahilhindwani-harness I need some clarity and confirmation on the technical accuracy of this point.

File: docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+## Important notes
+
+* Rollback deployment is currently supported for the following deployment types only: 
+  - Kubernetes
+  - Tanzu Application Services (TAS)
+
+    Harness anticipates expanding this feature to other deployment types in the future.
+* You can rollback successful pipelines only. The rollback option is not available for failed pipelines.
+* Only the rollback steps that are part of the stage can be rolled back.
+* You cannot rollback the same pipeline multiple times. It can be rolled back if the pipeline is executed again.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md:24]:
```suggestion
* You cannot rollback the same pipeline multiple times. You can perform rollback if the pipeline is executed again.
```

File: docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md (hunk 1 above)
Comment (sahilhindwani-harness) [reply docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md:26]: Looks good to me

Comment (sahilhindwani-harness) [reply docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md:26]: We can refer an expression of normal step executed during pipeline execution in the rollback section as well and it should be automatically resolved with that value from original execution incase of post prod rollback. 

Maybe we can write the above as our last point.

File: docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* Rollback deployment is currently supported for the following deployment types only: 
+  - Kubernetes
+  - Tanzu Application Services (TAS)
+
+    Harness anticipates expanding this feature to other deployment types in the future.
+* You can rollback successful pipelines only. The rollback option is not available for failed pipelines.
+* Only the rollback steps that are part of the stage can be rolled back.
+* You cannot rollback the same pipeline multiple times. You can perform rollback if the pipeline is executed again.
+* If the pipeline configuration has changed between executions, the previous execution YAML is used as reference to rollback.
+* If the rollback steps of a stage have the same expressions present in the steps of a successful deployment, the steps of the deployed pipeline must be resolved first. For example, you should be able to reuse a step in the rollback section as the output of a step in the steps section.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md:26]:
```suggestion
* You can refer expressions used in a step executed during the pipeline execution in the deployment rollback step as well. The expressions are automatically resolved with values from the original execution in the deployment rollback step.
```

**Review by sahilhindwani-harness — APPROVED** [review]

**Review by SudheendraKatte — DISMISSED** [review]
Approved with a minor suggestion.

**Pull request merged**